File: python/insertionSort.py
def insertionSort(unSortedList):
  iteration=1
  for i in range(1, len(unSortedList)):
     key = unSortedList[i]
     print("working on key",key)
     while(unSortedList[i-1]>key and i>0):
          unSortedList[i]=unSortedList[i-1]
          i = i-1
     unSortedList[i]=key
     print("after %s iteration"%iteration, unSortedList)
     iteration = iteration + 1
  print("after sorting completely", unSortedList)

          
     





unSortedList=[9,1,3,11,7,4,2,8,10,5,6]
print("numbers before sorting", unSortedList)
insertionSort(unSortedList)

File: insertionSort.c
#include<stdio.h>

printMyArray(int *a, int size)
{

for(int i = 0; i<size; i++)
  {
  printf("%d\t",a[i]);
  }

}


insertionSort(int *a)
{
 printf("\napplying insertion sort\n");

for(int i=1;i<11;i++)
  {
  int key = a[i];
  printf("working on element %d \n",key);
  while(key < a[i-1] && i>0)
      {
      a[i]=a[i-1];
      i = i-1;
      }
   a[i]=key;
  printf("after %d iteration\n",i);
  printMyArray(a,11);
  }
}


main()
{

int a[11]={9,1,3,11,7,4,2,8,10,5,6};

printf("numbers before sorting:\n");

printMyArray(a,11);

insertionSort(a);

printf("\n after sorting\n");

printMyArray(a,11);


}
